external/js/cothority/src/byzcoin/client-transaction.ts

```diff
@@ -364,6 +366,73 @@ export class Instruction extends Message<Instruction> {
         });
         return h.digest();
     }
+
+    /**
+     * Return a general-purpose json representation of an instruction that is
+     * contracts aware, ie. knows how to render specific contracts arguments.
+     * 
+     * The JSON has the following structure:
+     * 
+     *  {
+     *      type: "spawn|invoke|delete",
+     *      contract: "config|darc|..."
+     *      args: [
+     *          {
+     *              "name": "<NAME>",
+     *              "value": "<VALUE>",    // Arg value or its summary
+     *              "full": "<FULL VALUE>" // Optional. useful if the argument
+     *                                     // have a very long representation.
+     *          }
+     *          ...
+     *      ]
+     *  }
+     * 
+     */
+    getArranged(): { [k: string]: any } {
```

Comment:
The return type should be something more restrictive. From a quick look, I see that all returns contain the `contract` and `args` field.